src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternMatcher.cs

```diff
@@ -287,10 +287,9 @@ private bool SavePathSegmentsAsValues(int index, RouteValueDictionary values, St
             return false;
         }
 
-        private bool MatchComplexSegment(
+        internal static bool MatchComplexSegment(
```

Comment:
This is the existing code for processing complex segments. If we need to at some point we can work on the perf of it (I'm sure it's bad). As it is though it just works with the DFA matcher